cfme/tests/cloud/test_keypairs.py

```diff
@@ -93,3 +94,76 @@ def test_keypair_create_cancel(openstack_provider):
 
     with pytest.raises(KeyPairNotFound):
         navigate_to(keypair, 'Details')
+
+
+def test_keypair_add_tag(openstack_provider):
+    """ This will test whether it will add tag for newly created Keypair or not and then deletes it.
+
+    Steps:
+        * Provide Keypair name.
+        * Select Cloud Provider.
+        * Add tag to Keypair.
+        * Also delete it.
+    """
+    tag = ('Department', 'Accounting')
+    keypair = KeyPair(name=fauxfactory.gen_alphanumeric(), provider=openstack_provider)
+    try:
+        keypair.create()
+    except TimedOutError:
+        if BZ(1444520, forced_streams=['5.6', '5.7', 'upstream']).blocks:
+            pytest.skip('Timed out creating keypair, BZ1444520')
+        else:
+            pytest.fail('Timed out creating keypair')
+    view = navigate_to(keypair, 'All')
+    assert view.is_displayed
+
+    keypair.add_tag(tag)
+    tagged_value = mixins.get_tags(tag="My Company Tags")
+    assert tagged_value == ["Department: Accounting"], "Add tag failed."
```

Comment:
**Optional**
Manipulate original `tag` tuple to create the list for comparison here, so which tag is tested is easier to change/parametrize/etc.